### PR TITLE
Fix integer overflow in tf.range

### LINKED ISSUE #64081: slient overflow occurs in tf.range leading to incorrect result, here is a possible fix
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.15.0

### Custom code

Yes

### OS platform and distribution

_No response_

### Mobile device

_No response_

### Python version

_No response_

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

Following the documentation https://www.tensorflow.org/api_docs/python/tf/range, tf.range should have consistent result with np.arange. 
However, given the following inputs, tf.range outputs `tf.Tensor([1136033460], shape=(1,), dtype=int32)`, which is **incorrect**
The correct result is np.arange's output `[1136033460 -713794229]`.

I did some analysis and find the root cause lies in here:
https://github.com/tensorflow/tensorflow/blob/06e37b8d7f2ba0338efd0f588f068122414f0722/tensorflow/core/kernels/sequence_ops.cc#L100

When computing the `limit - start`, the actual value overflows.
To solve this problem, we can change the original code to a new one by applying the division first
origin:
```
Eigen::numext::ceil(Eigen::numext::abs((limit - start) / delta));
```

new:
```
Eigen::numext::ceil(Eigen::numext::abs(limit / delta - start / delta);
```

### Standalone code to reproduce the issue

```shell
import numpy as np
import tensorflow as tf
start = 1136033460
end = -2110457150
step = -1849827689
out = tf.range(start, end, step)
print(out)
out_np = np.arange(start, end, step)
print(out_np)
```
```


### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mattbahr**
Fixes #64081

### CODE REVIEW

File: tensorflow/core/kernels/sequence_ops.cc [hunk 1]
```diff
@@ -93,8 +93,16 @@ class RangeOp : public OpKernel {
     }
     int64_t size;
```
Comment (mattbahr) [inline tensorflow/core/kernels/sequence_ops.cc:94]: @cantonios Should we cast size as a `uint64_t`?

Comment (cantonios) [reply tensorflow/core/kernels/sequence_ops.cc:94]: Do you mean "declare"?  Probably.

Comment (mihaimaruseac) [reply tensorflow/core/kernels/sequence_ops.cc:94]: +1 on declaring it as that

**Review by mihaimaruseac — APPROVED** [review]
I think this looks good.

File: tensorflow/core/kernels/sequence_ops.cc (hunk 1 above)
Comment (mattbahr) [reply tensorflow/core/kernels/sequence_ops.cc:94]: ok, i can make the change when i'm at my computer this afternoon

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mattbahr** [comment]
@mihaimaruseac @cantonios Would either of you expect this change to impact the API compatibility? I've been trying to figure out why the `feedback/copybara` check was failing. The API compatibility test fails when I run it locally, and when I go to update goldens, I'm getting changes to a lot of files that I wouldn't have expected based on my changes as they seem unrelated.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
> @mihaimaruseac @cantonios Would either of you expect this change to impact the API compatibility? I've been trying to figure out why the `feedback/copybara` check was failing. The API compatibility test fails when I run it locally, and when I go to update goldens, I'm getting changes to a lot of files that I wouldn't have expected as they seem unrelated to my changes.

That looks weird, there should be no change in API.

**Comment by mihaimaruseac** [comment]
It's an issue with strict dependencies (BUILD rules): 

```
ERROR: Strict deps violations: .../tensorflow/python/ops/ragged:ragged_range_op_test
  In .../tensorflow/python/ops/ragged/ragged_range_op_test.py, no direct deps found for imports:
    line 16: import numpy as np: Module not provided by a direct dep
```

**Comment by mihaimaruseac** [comment]
In https://github.com/tensorflow/tensorflow/blob/2c5391cdedea3e970856ce6b5cf52b239620bdbf/tensorflow/python/ops/ragged/BUILD#L895-L908, you need to add `//third_party/py/numpy`

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mattbahr** [comment]
Hi @mihaimaruseac is there anything holding up this PR at this point? Or is it just waiting on resources to finish the checks and merge?

**Comment by mihaimaruseac** [comment]
Hi. It needs to be reviewed internally by someone other than me.

**Comment by mihaimaruseac** [comment]
CC @gbaned

**Comment by mihaimaruseac** [comment]
This is getting rolled back due to it introducing a signed integer overflow issue.

File: tensorflow/core/kernels/sequence_ops.cc [hunk 2]
```diff
@@ -91,18 +91,26 @@ class RangeOp : public OpKernel {
...
+      uint64_t range;
+      if ((limit > 0 && start < 0) || (limit < 0 && start > 0)) {
+        range = static_cast<uint64_t>(Eigen::numext::abs(limit)) 
+                + static_cast<uint64_t>(Eigen::numext::abs(start));
+      } else {
+        range = static_cast<uint64_t>(Eigen::numext::abs(limit - start)); 
+      }
+
+      size = Eigen::divup(range,
+                          static_cast<uint64_t>(Eigen::numext::abs(delta)));
```
Comment (mattbahr) [inline tensorflow/core/kernels/sequence_ops.cc:105]: @mihaimaruseac I think we should add a check here that `size` is less than or equal to `std::numeric_limits<int64_t>::max()`.

**Pull request merged**